pkg/controller/replication/replication_controller.go

```diff
@@ -456,6 +507,48 @@ func (rm *ReplicationManager) worker() {
 	}
 }
 
+func (rm *ReplicationManager) removeControllerRef(task abandon) error {
```

Comment:
It's probably not worth to do here, but I'm thinking that sometime in the future we might want to have a generic "removeFromList" function for API objects. @lavalamp